### PR TITLE
REL-2758 default scheduling blocks

### PR DESCRIPTION
**Pull request by tpolecat**
New observations created in the OT will now have a default scheduling block set to the middle of the program semester, if any, or the middle of the current semester, whichever is later.

### CODE REVIEW

File: bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/util/DefaultSchedulingBlock.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+import edu.gemini.pot.sp.ISPNode
+import edu.gemini.spModel.core.{Site, Semester, ProgramId}
+import edu.gemini.spModel.obs.SchedulingBlock
+
+import scalaz._, Scalaz._
+
+// N.B. this causes the object methods to appear as static methods on DefaultSchedulingBlock
+// which makes it easier to call from Java.
+sealed abstract class DefaultSchedulingBlock
```
Comment (cquiroz) [inline bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/util/DefaultSchedulingBlock.scala]: So it avoids making the call like `DefaultSchedulingBlock$.MODULE$`? Neat

File: bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/util/DefaultSchedulingBlock.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+object DefaultSchedulingBlock {
+
+  /**
+   * Construct a default scheduling block for program in which `node` lives. This block will
+   * start at the center of the program semester OR the current semester, whichever is later,
+   * and will have no duration.
+   */
+  def forProgram(node: ISPNode): SchedulingBlock =
+    getSchedulingBlock(node, Site.GS)
```
Comment (cquiroz) [inline bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/util/DefaultSchedulingBlock.scala]: Why does it default to `GS`?

Comment (tpolecat) [inline bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/util/DefaultSchedulingBlock.scala]: Coin flip. The point is just to get you near the middle of the semester, which is a point in time that's a few hours apart depending on the site. Doesn't matter here.

File: bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/util/DefaultSchedulingBlock.scala (hunk 1 above)
Comment (tpolecat) [inline bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/util/DefaultSchedulingBlock.scala]: Yeah, I didn't know this until recently. It's a nice trick.

**Comment by cquiroz** [comment]
Looks good to me :+1:

**Pull request merged**